flink-staging/flink-ml/src/main/scala/org/apache/flink/ml/statistics/ContinuousHistogram.scala

```diff
@@ -0,0 +1,315 @@
+/*
+ * Licensed to the Apache Software Foundation (ASF) under one
+ * or more contributor license agreements.  See the NOTICE file
+ * distributed with this work for additional information
+ * regarding copyright ownership.  The ASF licenses this file
+ * to you under the Apache License, Version 2.0 (the
+ * "License"); you may not use this file except in compliance
+ * with the License.  You may obtain a copy of the License at
+ *
+ *     http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.apache.flink.ml.statistics
+
+import scala.Double.MaxValue
+import scala.collection.mutable
+
+/** Implementation of a continuous valued online histogram
+  * Adapted from Ben-Haim and Yom-Tov's Streaming Decision Tree Algorithm
+  * Refer http://www.jmlr.org/papers/volume11/ben-haim10a/ben-haim10a.pdf
+  *
+  * =Parameters=
+  * -[[capacity]]:
+  * Number of bins to be used in the histogram
+  *
+  * -[[min]]:
+  * Lower limit on the elements
+  *
+  * -[[max]]:
+  * Upper limit on the elements
+  */
+class ContinuousHistogram(capacity: Int, min: Double, max: Double) extends OnlineHistogram {
+
+  private val lower = min
+  private val upper = max
+
+  require(capacity > 0, "Capacity should be a positive integer")
+  require(lower < upper, "Lower must be less than upper")
+
+  val data = new mutable.ArrayBuffer[(Double, Int)]()
```

Comment:
Using an `ArrayBuffer` for the data storage is highly inefficient. Insert and remove operations have `O(n)` complexity. Since you perform these operations very often, it will be painfully slow. A better solution would be to use a `TreeSet` for example. A `TreeSet's` insert, remove and floor/ceiling function have complexity of `O(log(n))`.